crates/libs/core/src/ref_mut.rs

```diff
@@ -0,0 +1,25 @@
+use super::*;
+
+/// A borrowed type with the same memory layout as the type itself that can be used to construct ABI-compatible function signatures.
+///
+/// This is a mutable version of [Ref] meant to support out parameters.
```

Comment:
What would you think of renaming `RefMut` to `OutRef`?

Reply:
I'm trying to borrow 😏 language from Rust where practical. There's `Param` and `ParamMut` and now `Ref` and `RefMut` which mirrors similar naming in the Rust Standard Library. `RefMut` and `ParamMut` take care of all the out parameter conventions, lifetimes, and stack packing rules required by stdcall, Win32, COM, but behaves like `&mut` in Rust.

Reply:
Ref and RefMut in the standard library do something different, though.  For example, std::cell::Ref and std::cell::RefMut are both lifetime-tracked pointers a an element in a cell.  The level of the pointer is the same, and they only differ based on whether you've acquired shared or exclusive access to the thing inside the cell.

But in this case, Ref is a pointer to an object, but RefMut is a pointer-to-a-pointer to an object.  And they serve quite different purposes; one is intended as an input parameter, one as an output parameter.

Reply:
That's fair. I don't disagree. Naming is hard... 🥹

In that case I would probably go with `Param` and `OutParam` for the traits and `Ref` and `OutRef` for the reference wrappers as you suggest.

Reply:
done!